Data/Hashable/Class.hs

```diff
@@ -826,6 +826,10 @@ instance Hashable1 Proxy where
 instance Hashable a => Hashable (NE.NonEmpty a) where
     hashWithSalt p (a NE.:| as) = p `hashWithSalt` a `hashWithSalt` as
 
+-- | @since UNRELEASED
+instance Hashable1 NE.NonEmpty where
+    liftHashWithSalt h salt (a NE.:| as) = liftHashWithSalt h (h salt a) as
+
 instance Hashable a => Hashable (Min a) where
```

Comment:
I'd add instances to all `semigroups` types. `Min`, `Max` ...

And also to types from `Data.Monoid`: `All`, `Any`, `First`, `Last`, `Sum`, `Product`, `Alt`, `Ap`...

Reply:
I agree that these would be nice to have too.

I can't promise I'll work on this though.